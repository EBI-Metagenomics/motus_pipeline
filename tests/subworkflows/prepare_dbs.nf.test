nextflow_process {

    name "Test SubWorkflow for download DBs"
    script "subworkflows/prepare_dbs.nf"
    
    process "DOWNLOAD_MAPSEQ_LSU"
    test("find existing mapseq silva fake DB") {
        when {
            params {
                databases = "tests/subworkflows/fixtures/db"
                silva_lsu_db_name = "silva"
            }
            process {
                """
                """
            }
        }
        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }
    
    process "DOWNLOAD_MAPSEQ_LSU"
    test("download DB for mapseq LSU") {
        when {
            params {
                databases = "tests/subworkflows/fixtures/db"
            }
            process {
                """
                """
            }
        }
        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }
}
