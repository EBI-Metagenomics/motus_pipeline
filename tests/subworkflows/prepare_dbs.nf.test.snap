{
    "download DB": {
        "content": [
            {
                "mapseq_db_lsu": [
                    [
                        "LSU.fasta:md5,96dc05a4ab2933ec6f443a1a0ce0d225",
                        "LSU.fasta.mscluster:md5,19f454e601925abbde17302ddc20c165",
                        "lsu2.otu:md5,515df01f33ed8d5741fbe0e55e4780b9",
                        "slv_lsu_filtered2.txt:md5,d521ce911cb9b94f4d304286ed9f8c3e"
                    ]
                ]
            }
        ],
        "timestamp": "2023-03-23T13:27:05+0000"
    },
    "find existing mapseq silva fake DB": {
        "content": [
            {
                "mapseq_db_lsu": [
                    "silva:md5,1d2270ac6c6024ec551dd152a6230530"
                ]
            }
        ],
        "timestamp": "2023-03-23T13:27:02+0000"
    }
}